pkgs/development/compilers/gcc/builder.sh

```diff
@@ -213,9 +213,24 @@ preInstall() {
 
 postInstall() {
     # Move runtime libraries to $lib.
-    moveToOutput "lib/lib*.so*" "$lib"
-    moveToOutput "lib/lib*.la"  "$lib"
-    moveToOutput "lib/lib*.dylib" "$lib"
+    if [[ -d "$out/$target_triple/lib" && -n "$(ls -A $out/$target_triple/lib)" ]]; then
+        moveToOutput "$target_triple/lib/lib*.so*" "$lib"
+        moveToOutput "$target_triple/lib/lib*.la" "$lib"
+        moveToOutput "$target_triple/lib/lib*.dylib" "$lib"
+        mv "$lib/$target_triple"/lib/* "$lib/lib/"
+        rmdir -p --ignore-fail-on-non-empty "$lib/$target_triple/lib" || :
+    elif [[ -d "$out/$target_triple/lib64" && -n "$(ls -A $out/$target_triple/lib64)" ]]; then
+        moveToOutput "$target_triple/lib64/lib*.so*" "$lib"
```

Comment:
I think you can remove the special handling of `lib64` if you replace `$out` with `$out/$targetConfig` in the `preInstall` hook above. This should cause the libraries to be installed into `lib` rather than `lib64`, and since `$targetConfig` is empty during native builds, it will work for both native and cross.